src/worker/vm/lazy.ts

```diff
@@ -14,38 +15,72 @@ import { clearError, processError } from '../../utils/db/error'
 import { disablePlugin, setPluginCapabilities } from '../../utils/db/sql'
 import { status } from '../../utils/status'
 import { createPluginConfigVM } from './vm'
-
 export class LazyPluginVM {
     initialize?: (hub: Hub, pluginConfig: PluginConfig, indexJs: string, logInfo: string) => Promise<void>
     failInitialization?: () => void
-    resolveInternalVm: Promise<PluginConfigVMResponse | null>
+    resolveInternalVm!: Promise<PluginConfigVMResponse | null>
+    totalAttemptsToInitialize: number
 
     constructor() {
+        this.totalAttemptsToInitialize = 0
+        this.initVM()
+    }
+
+    private initVM() {
+        this.totalAttemptsToInitialize++
         this.resolveInternalVm = new Promise((resolve) => {
             this.initialize = async (hub: Hub, pluginConfig: PluginConfig, indexJs: string, logInfo = '') => {
-                try {
-                    const vm = await createPluginConfigVM(hub, pluginConfig, indexJs)
+                const createPluginLogEntry = async (
+                    message: string,
+                    logType = PluginLogEntryType.Info
+                ): Promise<void> => {
                     await hub.db.createPluginLogEntry(
                         pluginConfig,
                         PluginLogEntrySource.System,
-                        PluginLogEntryType.Info,
-                        `Plugin loaded (instance ID ${hub.instanceId}).`,
+                        logType,
+                        message,
                         hub.instanceId
                     )
+                }
+                try {
+                    const vm = await createPluginConfigVM(hub, pluginConfig, indexJs)
+                    await createPluginLogEntry(`Plugin loaded (instance ID ${hub.instanceId}).`)
                     status.info('🔌', `Loaded ${logInfo}`)
                     void clearError(hub, pluginConfig)
                     await this.inferPluginCapabilities(hub, pluginConfig, vm)
                     resolve(vm)
                 } catch (error) {
-                    await hub.db.createPluginLogEntry(
-                        pluginConfig,
-                        PluginLogEntrySource.System,
-                        PluginLogEntryType.Error,
-                        `Plugin failed to load and was disabled (instance ID ${hub.instanceId}).`,
-                        hub.instanceId
+                    const isRetryError = error instanceof RetryError
+                    status.warn('⚠️', error.message)
+                    if (isRetryError && this.totalAttemptsToInitialize < 15) {
+                        const nextRetryMs = 2 ** this.totalAttemptsToInitialize * 3000
```

Comment:
Suggestion: `INITIALIZATION_RETRY_MULTIPLIER ** (this.totalAttemptsToInitialize - 1) * INITIALIZATION_RETRY_BASE_MS`. That way we'd have the initial retry timeout and its multiplier plainly in constants, as right know it's not obvious that the first retry will be 6000 ms later.